src/internal/connector/graph/uploadsession.go

```diff
@@ -29,36 +28,36 @@ type writer struct {
 	contentLength int64
 	// Last item offset that was written to
 	lastWrittenOffset int64
-	client            *resty.Client
+	client            httpWrapper
 }
 
-func NewWriter(id, url string, size int64) *writer {
-	return &writer{id: id, url: url, contentLength: size, client: resty.New()}
+func NewLargeItemWriter(id, url string, size int64) *largeItemWriter {
+	return &largeItemWriter{id: id, url: url, contentLength: size, client: *NewNoTimeoutHTTPWrapper()}
 }
 
 // Write will upload the provided data to M365. It sets the `Content-Length` and `Content-Range` headers based on
 // https://docs.microsoft.com/en-us/graph/api/driveitem-createuploadsession
-func (iw *writer) Write(p []byte) (int, error) {
+func (iw *largeItemWriter) Write(p []byte) (int, error) {
 	rangeLength := len(p)
-	logger.Ctx(context.Background()).
+	ctx := context.Background()
+
+	logger.Ctx(ctx).
 		Debugf("WRITE for %s. Size:%d, Offset: %d, TotalSize: %d",
 			iw.id, rangeLength, iw.lastWrittenOffset, iw.contentLength)
 
 	endOffset := iw.lastWrittenOffset + int64(rangeLength)
 
 	// PUT the request - set headers `Content-Range`to describe total size and `Content-Length` to describe size of
 	// data in the current request
-	_, err := iw.client.R().
-		SetHeaders(map[string]string{
-			contentRangeHeaderKey: fmt.Sprintf(
-				contentRangeHeaderValueFmt,
-				iw.lastWrittenOffset,
-				endOffset-1,
-				iw.contentLength),
-			contentLengthHeaderKey: fmt.Sprintf("%d", rangeLength),
-		}).
-		SetBody(bytes.NewReader(p)).
-		Put(iw.url)
+	headers := make(map[string]string)
+	headers[contentRangeHeaderKey] = fmt.Sprintf(
+		contentRangeHeaderValueFmt,
+		iw.lastWrittenOffset,
+		endOffset-1,
+		iw.contentLength)
+	headers[contentLengthHeaderKey] = fmt.Sprintf("%d", rangeLength)
+
+	_, err := iw.client.Request(ctx, "PUT", iw.url, bytes.NewReader(p), headers)
```

Comment:
Always prefer http library consts for http values, please.
```suggestion
	_, err := iw.client.Request(ctx, http.MethodPut, iw.url, bytes.NewReader(p), headers)
```